include/net/nrf_cloud_coap.h

```diff
@@ -0,0 +1,350 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef NRF_CLOUD_COAP_H_
+#define NRF_CLOUD_COAP_H_
+
+/** @file nrf_cloud_coap.h
+ * @brief Module to provide nRF Cloud CoAP API
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <net/nrf_cloud_rest.h>
+#include <net/nrf_cloud_agps.h>
+#include <net/nrf_cloud_pgps.h>
+#include <zephyr/net/coap_client.h>
+
+/**
+ * @defgroup nrf_cloud_coap nRF CoAP API
+ * @{
+ */
+
+/* Transport functions */
+/**@brief Initialize nRF Cloud CoAP library.
+ *
+ * @return 0 if initialization was successful, otherwise, a negative error number.
+ */
+int nrf_cloud_coap_init(void);
+
+/**@brief Connect to and obtain authorization to access the nRF Cloud CoAP server.
+ *
+ * Upon successful authorization, call @ref nrf_cloud_coap_is_connected to confirm
+ * whether the device was granted access to nRF Cloud.
+ *
+ * @return 0 if authorized successfully, otherwise, a negative error number.
+ */
+int nrf_cloud_coap_connect(void);
+
+/**@brief Check if device is connected and authorized to use nRF Cloud CoAP.
+ *
+ * A device is authorized if the JWT it POSTed to the /auth/jwt endpoint is valid
+ * and matches credentials for a device already provisioned and associated with nRF Cloud.
+ *
+ * @retval true Device is allowed to access nRF Cloud services.
+ * @retval false Device is disallowed from accessing nRF Cloud services.
+ */
+bool nrf_cloud_coap_is_connected(void);
```

Comment:
Wouldn't it be better if the application could subscribe to a callback to get notified if connected/authorized or not?
Then the app could simply switch state on such an occurance instead of constalty checking if connected prior to updating a resource.

Reply:
The function is defined in `nrf_cloud_coap_transport.c` and used in `nrf_cloud_coap.c` for all the functions to interact with various services.  These latter functions all call `nrf_cloud_coap_is_connected()` before proceeding, so it is not necessary for the user to call that function.

I can move it into a local header.